File: single-language-analysis/dataset-files/python/test_smell_153.py
# Original URL: https://github.com/code-refactor/schedualpy_fintech_maria/blob/8c9486e6c3b96e6137731514889bba90b5cd08ab/test_api.py#L22-L34

def test_smell_153(client):
    rv = client.post('/jobs/recurring', json={'interval': 0.1, 'sla_jitter': 0, 'func': 'y'})
    jid = rv.get_json()['job_id']
    time.sleep(0.3)
    rv2 = client.delete(f'/jobs/{jid}/cancel')
    assert rv2.status_code == 200
    rv3 = client.post(f'/jobs/{jid}/reschedule', json={'interval': 0.05})
    assert rv3.status_code == 200
    # Rescheduling a cancelled job should error or do nothing
    # We accept no exception; check response content
    data = rv3.get_json()
    assert data['rescheduled'] == jid
    assert data['new_interval'] == 0.05

File: single-language-analysis/dataset-files/java/TestSmell153.java
import org.junit.Test;
import static org.junit.Assert.*;

public class TestSmell153 {

    @Test
    public void testRecurringAndCancelAndReschedule() {
        Response rv = client.post("/jobs/recurring", json("{\"interval\": 0.1, \"sla_jitter\": 0, \"func\": \"y\"}"));
        String jid = rv.getJson().get("job_id").asText();
        try {
            Thread.sleep(300);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        Response rv2 = client.delete("/jobs/" + jid + "/cancel");
        assertEquals(200, rv2.getStatusCode());
        Response rv3 = client.post("/jobs/" + jid + "/reschedule", json("{\"interval\": 0.05}"));
        assertEquals(200, rv3.getStatusCode());
        JsonNode data = rv3.getJson();
        assertEquals(jid, data.get("rescheduled").asText());
        assertEquals(0.05, data.get("new_interval").asDouble(), 0.0);
    }
}